src/containers/team-leaderboard/TeamLeaderboard.tsx

```diff
@@ -0,0 +1,136 @@
+  readonly teamLeaderboardItemsWeekly: TeamLeaderboardWeeklyReducerState['teamLeaderboardWeekly'];
+  readonly teamLeaderboardItemsMonthly: TeamLeaderboardMonthlyReducerState['teamLeaderboardMonthly'];
+  readonly teamLeaderboardItemsYearly: TeamLeaderboardYearlyReducerState['teamLeaderboardYearly'];
+  readonly teamLeaderboardItemsAllTime: TeamLeaderboardAllTimeReducerState['teamLeaderboardAllTime'];
+}
+
+const TeamLeaderboard: React.FC<TeamLeaderboardProps> = ({
+  teamLeaderboardItemsWeekly,
+  teamLeaderboardItemsMonthly,
+  teamLeaderboardItemsYearly,
+  teamLeaderboardItemsAllTime,
+}) => {
+  const dispatch = useDispatch();
+
+  const [weeklyTab, setWeeklyTab] = useState<TabInfo>({
+    name: 'Weekly',
+    content: [],
+  });
+  const [monthlyTab, setMonthlyTab] = useState<TabInfo>({
+    name: 'Monthly',
+    content: [],
+  });
+  const [yearlyTab, setYearlyTab] = useState<TabInfo>({
+    name: 'Yearly',
+    content: [],
+  });
+  const [allTimeTab, setAllTimeTab] = useState<TabInfo>({
+    name: 'All Time',
+    content: [],
+  });
+
+  useEffect(() => {
+    dispatch(getTeamsLeaderboard(LeaderboardPreviousDays.weekly));
+    dispatch(getTeamsLeaderboard(LeaderboardPreviousDays.monthly));
+    dispatch(getTeamsLeaderboard(LeaderboardPreviousDays.yearly));
+    dispatch(getTeamsLeaderboard(LeaderboardPreviousDays.allTime));
+  }, []);
+
+  useEffect(() => {
+    teamLeaderboardItemsWeekly.kind === AsyncRequestKinds.Completed &&
+      setWeeklyTab({
+        ...weeklyTab,
+        content: leaderboardItemsToTabItems(teamLeaderboardItemsWeekly.result),
+      });
+  }, [teamLeaderboardItemsWeekly]);
+
+  useEffect(() => {
+    teamLeaderboardItemsMonthly.kind === AsyncRequestKinds.Completed &&
+      setMonthlyTab({
+        ...monthlyTab,
+        content: leaderboardItemsToTabItems(teamLeaderboardItemsMonthly.result),
+      });
+  }, [teamLeaderboardItemsMonthly]);
+
+  useEffect(() => {
+    teamLeaderboardItemsYearly.kind === AsyncRequestKinds.Completed &&
+      setYearlyTab({
+        ...yearlyTab,
+        content: leaderboardItemsToTabItems(teamLeaderboardItemsYearly.result),
+      });
+  }, [teamLeaderboardItemsYearly]);
+
+  useEffect(() => {
+    teamLeaderboardItemsAllTime.kind === AsyncRequestKinds.Completed &&
+      setAllTimeTab({
+        ...weeklyTab,
+        content: leaderboardItemsToTabItems(teamLeaderboardItemsAllTime.result),
+      });
+  }, [teamLeaderboardItemsAllTime]);
+
+  return (
+    <LeaderboardContentContainer>
+      <PageHeader
+        pageTitle="Team Leaderboard"
+        pageSubtitle="Let’s celebrate your team reaching for its goals!"
+      />
+
+      <LeaderboardContainer>
+        {teamLeaderboardItemsWeekly.kind === AsyncRequestKinds.Completed &&
+          teamLeaderboardItemsMonthly.kind === AsyncRequestKinds.Completed &&
+          teamLeaderboardItemsYearly.kind === AsyncRequestKinds.Completed &&
+          teamLeaderboardItemsAllTime.kind === AsyncRequestKinds.Completed && (
```

Comment:
Just pointing out, this implies that all four variations of leaderboard must be loaded in order to view one of them. 

@florisdobber / @liammoyn what does the backend for this do when given a number other than 7, 30, etc? If it's truly flexible I think we should make this code a lot smaller.